python/ray/train/lightning/lightning_trainer.py

```diff
@@ -0,0 +1,108 @@
+    a batch size for your datasets, you should set the ``batch_size`` key of
+    ``lightning_module_init_config``. Note that in your ``LightningModule``'s
+    ``__init__``, you should not accept a ``batch_size`` argument.
+
+    This Trainer requires ``pytorch-lightning>=1.7.0`` package.
+
+    Args:
+        lightning_module: A class object (not a class instance) that is a subclass
+            of ``pytorch_lightning.LightningModule``. This class should define your
+            model logic.
+        lightning_module_init_config: Configurations to pass into
+            ``lightning_module.__init__`` as kwargs.
+        trainer_init_config: Configurations to pass into
+            ``pytorch_lightning.Trainer.__init__`` as kwargs. For valid arguments to
+            pass, see
+            https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#init.
+        ddp_strategy_init_config: Configurations to pass into
+            ``pytorch_lightning.strategies.DDPStrategy.__init__`` as kwargs. Most users
+            should only set this to ``{"find_unused_parameters": False}`` or leave this
+            as-is. For valid arguments to pass, see
+            https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.strategies.DDPStrategy.html#pytorch_lightning.strategies.DDPStrategy
+            and
+            https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html#torch.nn.parallel.DistributedDataParallel.
+        torch_config: Configuration for setting up the PyTorch backend. If set to
+            None, use the default configuration. This replaces the ``backend_config``
+            arg of ``DataParallelTrainer``. Same as in ``TorchTrainer``.
+        scaling_config: Configuration for how to scale data parallel training.
+        dataset_config: Configuration for dataset ingest.
+        run_config: Configuration for the execution of the training run.
+        datasets: Any Ray Datasets to use for training. Use
+            the key "train" to denote which dataset is the training
+            dataset and (optionally) key "evaluation" to denote the evaluation
+            dataset. Can only contain a training dataset
+            and up to one extra dataset to be used for evaluation.
+            If a ``preprocessor`` is provided and has not already been fit,
+            it will be fit on the training dataset. All datasets will be
+            transformed by the ``preprocessor`` if one is provided.
+        preprocessor: A ray.data.Preprocessor to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+    """
+
+    def __init__(
+        self,
+        lightning_module: Type[pytorch_lightning.LightningModule],
+        *,
+        lightning_module_init_config: Optional[Dict] = None,
+        trainer_init_config: Optional[Dict] = None,
+        ddp_strategy_init_config: Optional[Dict] = None,
```

Comment:
Can we add more information to the type hints for these configs (`Dict[str, Any]`)